File: examples/python/04_multiarm.py
"""Simulated shared-control multi-arm CMR example."""

import numpy as np

import cmrdesign as cmr

rng = np.random.default_rng(104)
n_per_arm = 300
arm = np.repeat([0, 1, 2], n_per_arm)
y = np.r_[
    rng.beta(4, 4, n_per_arm),
    rng.beta(2, 6, n_per_arm),
    rng.beta(5, 3, n_per_arm),
]

fit = cmr.cmr_multiarm(y, arm, alpha=0.05, method="bounded")

print(fit.pi)
print({"certificate": fit.U_CMR})

File: examples/r/04_multiarm.R
# Simulated shared-control multi-arm CMR example.

library(cmrdesign)

set.seed(104)
n_per_arm <- 300L
arm <- rep(c(0, 1, 2), each = n_per_arm)
y <- c(
  rbeta(n_per_arm, 4, 4),
  rbeta(n_per_arm, 2, 6),
  rbeta(n_per_arm, 5, 3)
)

fit <- cmr_multiarm(y, arm, alpha = 0.05, method = "bounded")
allocation <- realize_allocation(fit, n_main = 1200)

print(fit$pi)
print(list(certificate = fit$U_CMR))
print(list(counts = allocation$counts,
           realized_certificate = allocation$realized_U_CMR))
